src/main/java/frc/robot/subsystems/gpm/Shooter.java

```diff
@@ -0,0 +1,199 @@
+
+    /**
+     * In RPM
+     */
+    public static final double TOLERANCE = 10;
+
+    private final CANSparkFlex bottomMotor = new CANSparkFlex(ShooterConstants.BOTTOM_MOTOR_ID, MotorType.kBrushless);
+    private final CANSparkFlex topMotor = new CANSparkFlex(ShooterConstants.TOP_MOTOR_ID, MotorType.kBrushless);
+    private final RelativeEncoder bottomMotorEncoder = bottomMotor.getEncoder();
+    private final RelativeEncoder topMotorEncoder = topMotor.getEncoder();
+    private final PIDController topPID = new PIDController(TOP_P, TOP_I, TOP_D);
+    private final PIDController bottomPID = new PIDController(BOTTOM_P, BOTTOM_I, BOTTOM_D);
+
+    // TODO: TUNE THIS
+    private final SimpleMotorFeedforward feedforward = new SimpleMotorFeedforward(S, V);
+
+    public Shooter() {
+        topPID.setTolerance(TOLERANCE);
+        bottomPID.setTolerance(TOLERANCE);
+        bottomMotor.setInverted(true);
+    }
+
+    @Override
+    public void periodic() {
+        double topSpeed = topMotorEncoder.getVelocity();
+        double bottomSpeed = bottomMotorEncoder.getVelocity();
+
+        topMotor.set(topPID.calculate(topSpeed) + feedforward.calculate(topPID.getSetpoint()));
+        bottomMotor.set(bottomPID.calculate(bottomSpeed) + feedforward.calculate(bottomPID.getSetpoint()));
+
+        SmartDashboard.putNumber("top speed", shooterRPMToSpeed(topSpeed));
+        SmartDashboard.putNumber("bottom speed", shooterRPMToSpeed(bottomSpeed));
+        SmartDashboard.putBoolean("at setpoint?", atSetpoint());
+        System.out.println(shooterRPMToSpeed(topSpeed) + "," + shooterRPMToSpeed(bottomSpeed));
+    }
+
+	/**
+	* Converts an RPM to a speed at the surface of the shooter wheels.
+	*
+	* @param  rpm the RPM to convert
+	* @return     the converted value in m/s
+	* @see        shooterSpeedToRPM
+	*/
+	public static double shooterRPMToSpeed(double rpm) {
+			return (rpm / 60) * (4 * Math.PI * 0.0254);
+	}
+
+	/**
+	* Converts a speed at the surface of the shooter wheels to an RPM value.
+	*
+	* @param  speed the speed to convert in m/s
+	* @return       the converted value in RPM
+	* @see          shooterRPMToSpeed
+	*/
+	public static double shooterSpeedToRPM(double speed) {
+			return (speed * 60) / (4 * Math.PI * 0.0254);
+	}
+
+	/**
+	* Sets the speed both shooter motors try to spin up to independantly.
+	*
+	* @param speedTop    the speed tho top motor will spin to in RPM
+	* @param speedBottom the speed the bottom motor will spin to in RPM
+	* @see               setTargetRPM
+	*/
+	public void setBothTargetRPMs(double speedTop, double speedBottom) {
+		topPID.setSetpoint(speedTop);
+		bottomPID.setSetpoint(speedBottom);
+	}
+
+	/**
+	* Sets the RPM both shooter motors try to spin up to.
+	*
+	* @param speed speed to spin to in RPM
+	* @see         setBothTargetRPMs
+	* @see         setTargetVelocity
+	*/
+    public void setTargetRPM(double speed) {
+        setBothTargetRPMs(speed * 0.5, speed);
+    }
```

Comment:
Why is the top motor spinning at 50% every time we shoot? This could mess up any calculations that need the note moving at a specific speed.

Reply:
we were testing whether exerting a torque that would cause the note to tilt up would help with shooting into the trap. for speaker, they should be the same speed.

Reply:
That was done for trap shooting tests.